File: HW6/5/complex.py
class complex:
	def __init__(self,real,imaginary):
		self.real = real
		self.imaginary = imaginary
	def getReal(self):
		return self.real
	def getImaginary(self):
		return self.imaginary
	def __add__(self,o):
		return complex(self.real + o.real, self.imaginary + o.imaginary)
	def __sub__(self,o):
		return complex(self.real - o.real, self.imaginary - o.imaginary)
	def __mul__(self,o):
		return complex(self.real * o.real, self.imaginary * o.imaginary)
	def __div__(self,o):
		return complex(self.real / o.real, self.imaginary / o.imaginary)
 	def __str__(self):
 		return str(self.real) + ", "  + str(self.imaginary) + "i"
a = complex(2,3)
b = a + a
c = b - a
d = a * b
e = b / a
print(a)
print(b)
print(c)
print(d)
print(e)
print(a.getReal())
print(a.getImaginary())

File: HW6/5/complex.cpp
#include <iostream>
using namespace std;
class complex{
	private:
		double real;
		double imaginary;
	public:
		complex(double r, double i){
			real = r;
			imaginary = i;
		}
		complex(complex a, complex b){
			real = a.real + b.real;
			imaginary = a.imaginary + b.imaginary;
		}
		complex(const complex& a){
			real = a.real;
			imaginary = a.imaginary;
		}
		double getReal(){
			return real;
		}
		double getImaginary(){
			return imaginary;
		}
		complex operator+(const complex other){
			complex temp(0,0);
			temp.real = real + other.real;
			temp.imaginary = imaginary + other.imaginary;
			return temp;
		}
		complex operator-(const complex other){
			complex temp(0,0);
			temp.real = real - other.real;
			temp.imaginary = imaginary - other.imaginary;
			return temp;
		}	
		complex operator*(const complex other){
			complex temp(0,0);
			temp.real = real * other.real;
			temp.imaginary = imaginary * other.imaginary;
			return temp;
		}	
		complex operator/(const complex other){
			complex temp(0,0);
			temp.real = real / other.real;
			temp.imaginary = imaginary / other.imaginary;
			return temp;
		}
		void print(){
    		printf( "%.4f + %.4fi", real, imaginary);
		}		
		
};
int main(){
	complex a(2,3);
	cout << "A real: " << a.getReal() << " A imaginary: " << a.getImaginary()<<endl;
	complex b = a + a;
	b.print();
	cout << endl;
	b = a-a-a;
	b.print();
	b = a*a;
	cout<<endl;
	b.print();
	b = a/a;
	cout<<endl;
	b.print();
	complex c(a,b);
	cout<<endl;
	c.print();
	complex d(a+b+c);
	cout<<endl;
	d.print();
}